### PR TITLE
cmake: fix std::filesystem detection and extract sanitizer detection into its own module

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by cbodley** [comment]
thanks @tchaikov, the filesystem change works for me with clang 5
```
$ cmake -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DWITH_SEASTAR=ON ..
...
-- Found Sanitizers: 1
-- Found StdFilesystem: stdc++fs
...
-- Found rt: rt
```
but adding `-DWITH_ASAN=ON` leads to this error in seastar:
```
-- Found Sanitizers: 1  
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find rt (missing: rt_LIBRARY_NAME)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  src/seastar/cmake/Findrt.cmake:47 (find_package_handle_standard_args)
  src/CMakeLists.txt:319 (_find_package)
  src/seastar/CMakeLists.txt:142 (find_package)
```
this may be a bug in Findrt.cmake? the first `try_compile()` succeeds without `LINK_LIBRARIES rt`:
> CMakeCache.txt:_rt_test_nil:INTERNAL=TRUE

but the `find_package_handle_standard_args()` uses `REQUIRED_VARS rt_LIBRARY_NAME`, which isn't set in this case (because it's not needed)

**Comment by tchaikov** [comment]
@cbodley fixed and repushed.

File: CMakeLists.txt [hunk 1]
```diff
@@ -558,43 +558,30 @@ endif(LINUX)
 
 # ASAN and friends
 option(WITH_ASAN "build with ASAN" OFF)
-option(WITH_ASAN_LEAK "explicitly enable ASAN leak detection" OFF)
-
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply CMakeLists.txt]: done.

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-kefu-testing-2018-08-03-0109/e3709e8a32e45d7c7d7589b7982d710502bf11c9/

**Pull request merged**